### PR TITLE
create a pipelines table in dynamodb

### PR DESCRIPTION
**Pull request by FenrirUnbound**
Initial PoC of creating a pipelines table in DynamoDB. 

What this PR does and does _not_ accomplish:
:white_check_mark:  create pipelines table
:negative_squared_cross_mark: document usability
:negative_squared_cross_mark:  define the final end-user module

### CODE REVIEW

File: config/pipeline.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+module.exports = {
+    AttributeDefinitions: [ /* required */
```
Comment (d2lam) [inline config/pipeline.js:3]: So these would include the required attributes?

File: lib/bobby.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+const AWS = require('aws-sdk');
+const CONFIGS = {
+    pipelines: require('../config/pipeline') // eslint-disable-line global-require
+};
+const dydb = new AWS.DynamoDB({
+    region: 'us-west-2'
+});
+
+/**
```
Comment (nkatzman) [inline lib/bobby.js]: Do you intend to give javadoc comments?

File: lib/bobby.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+const AWS = require('aws-sdk');
+const CONFIGS = {
+    pipelines: require('../config/pipeline') // eslint-disable-line global-require
+};
+const dydb = new AWS.DynamoDB({
+    region: 'us-west-2'
```
Comment (nkatzman) [inline lib/bobby.js]: Is this going to eventually be configurable via the command line?

Comment (FenrirUnbound) [inline lib/bobby.js]: It's now configurable.

This is to give the function a default region. The function caller can override it on a per-call basis.

File: lib/bobby.js (hunk 2 above)
Comment (FenrirUnbound) [inline lib/bobby.js]: Added!

File: config/pipeline.js (hunk 1 above)
Comment (FenrirUnbound) [inline config/pipeline.js:3]: Or at least the columns that we would index. There is planned work to be able to extend the `data-schema` so that we don't need to hard-code it here. This is included just so we can get an initial attempt working.

File: config/pipeline.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    ],
+    ProvisionedThroughput: { /* required */
+        ReadCapacityUnits: 1, /* required */
+        WriteCapacityUnits: 1 /* required */
+    },
+    TableName: 'Pipelines', /* required */
+    GlobalSecondaryIndexes: [
+        {
+            IndexName: 'ScmUrls', /* required */
```
Comment (d2lam) [inline config/pipeline.js:26]: Any reason why scmUrl is uppercase here? Is that the convention for index name?

File: config/pipeline.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            KeySchema: [ /* required */
+                {
+                    AttributeName: 'scmUrl',
+                    KeyType: 'HASH'
+                }
+            ],
+            Projection: { /* required */
+                ProjectionType: 'KEYS_ONLY'
```
Comment (d2lam) [inline config/pipeline.js:34]: what is this for?

File: lib/bobby.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+const AWS = require('aws-sdk');
+const CONFIGS = {
+    pipelines: require('../config/pipeline') // eslint-disable-line global-require
+};
+
+class Bobby {
+    /**
+     * [constructor description]
```
Comment (d2lam) [inline lib/bobby.js]: description?

File: lib/bobby.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+     * @param  {[type]} region =             'us-west-2' [description]
+     * @return {[type]}        [description]
```
Comment (d2lam) [inline lib/bobby.js]: add comments

File: config/pipeline.js (hunk 4 above)
Comment (FenrirUnbound) [inline config/pipeline.js:26]: Convenience of debugging.

The index is on SCM Urls. Take the scenario where we label the index name as "scmUrl". If the error states that "scmUrl" is invalid, then what's the issue: the index or the column?

Comment (FenrirUnbound) [inline config/pipeline.js:26]: There hasn't been much discussion regarding the configuration files. There is also planned work to formalize these configurations from the `data-schema`, so it was deemed that the work would actually be invested there.

This is to get the PoC in a place where others can contribute.

File: config/pipeline.js (hunk 5 above)
Comment (FenrirUnbound) [inline config/pipeline.js:34]: https://aws.amazon.com/dynamodb/faqs/

> Q: Can I retrieve all attributes of a DynamoDB table from a global secondary index?
> 
> A query on a GSI can only return attributes that were specified to be included in the GSI at creation time. The attributes included in the GSI are those that are projected by default such as the GSI’s key attribute(s) and table’s primary key attribute(s), and those that the user specified to be projected. For this reason, a GSI query will not return attributes of items that are part of the table, but not included in the GSI. A GSI that specifies all attributes as projected attributes can be used to retrieve any table attributes. See here for documentation on using GSIs for queries.

TL;DR - Only the attributes projected in the index will be available when queried. The initial pass was that we only care for uniqueness of `scmUrls` and necessarily any other data.

**Comment by d2lam** [comment]
Update README?

**Comment by FenrirUnbound** [comment]
README updated

EDIT: word

**Comment by d2lam** [comment]
👍
